docs/source/plugins/index.md

```diff
@@ -0,0 +1,11 @@
+(plugins-index)=
+
+# JupyterHub Pytest Plugins
+
+This document contains autogenerated documentation of individual plugin modules and their fixtures.
```

Comment:
Let's link the mention of plugin, fixtures to their respective documentation in pytest.
Also, let's mention the use of sphinx.ext.autodoc (and link to its docs) for the generation of this section.